internal/tasks/manifests_test.go

```diff
@@ -643,4 +643,21 @@ func TestCheckVulnerabilitiesForNextManifest(t *testing.T) {
 		UPDATE manifests SET next_vuln_check_at = 6120 WHERE repo_id = 1 AND digest = '%s';
 		UPDATE manifests SET next_vuln_check_at = 9600, vuln_status = 'Clean' WHERE repo_id = 1 AND digest = '%s';
 	`, images[0].Manifest.Digest, images[2].Manifest.Digest, images[1].Manifest.Digest)
+
+	// check retry on transient errors
+	imageError := test.GenerateImage(test.GenerateExampleLayer(3))
+	imageError.MustUpload(t, s, fooRepoRef, "")
+	tr.DBChanges().Ignore()
+
+	claird.IndexFixtures[imageError.Manifest.Digest.String()] = "fixtures/clair/manifest-004.json"
+	simulatedErrorMsg := "failed to scan all layer contents: failed to connect to host=clair-postgresql user=postgres database=clair: dial error (dial tcp 10.30.50.60:5432: connect: connection refused)"
+	mustExec(t, s.DB, `UPDATE manifests SET vuln_status = 'Error', vuln_scan_error = $1 where digest = $2`, simulatedErrorMsg, imageError.Manifest.Digest)
+
+	expectSuccess(t, j.CheckVulnerabilitiesForNextManifest())
+	expectError(t, sql.ErrNoRows.Error(), j.CheckVulnerabilitiesForNextManifest())
+
+	tr.DBChanges().AssertEqualf(`
+		UPDATE blobs SET blocks_vuln_scanning = FALSE WHERE id = 9 AND account_name = 'test1' AND digest = '%[2]s';
+		UPDATE manifests SET next_vuln_check_at = 6120 WHERE repo_id = 1 AND digest = '%[1]s';
+	`, imageError.Manifest.Digest, imageError.Layers[0].Digest)
```

Comment:
This should verify that the broken index report was deleted in Clair. In fact, something seems to be wrong here. The ClairDouble does not handle DELETE requests, so it should respond 405 (Method Not Allowed) when given a DELETE request. This all smells like the DELETE is not actually taking place.

I'd say you need to:

- extend the mock to handle DELETE requests according to its internal model, and
- have the mock generate the broken index report as a first step, ingest that through the vuln check, than swap out the fixture to generate a valid index report in the second step